src/__tests__/parser.ts

```diff
@@ -0,0 +1,730 @@
+import * as kolint from '..'
+import * as assert from 'assert'
+
+function parse(code: string) {
+	let document = kolint.parse('inline', code, kolint.createProgram())
+	// Serialize and parse to deep convert all classes to objects
+	document = JSON.parse(JSON.stringify(document)) as typeof document
+	return document
+}
+
+describe('Parser', () => {
+
+	describe('Elements', () => {
+
+		it('Normal tag', () => {
+			assert.deepStrictEqual(
+				parse('<div></div>'),
+				[
+					{
+						loc: {
+							first_line: 1,
+							last_line: 1,
+							first_column: 0,
+							last_column: 5,
+							range: [0, 5]
+						},
+						key: 'div',
+						nodeType: 0
+					},
```

Comment:
The test file becomes too verbose. It is hard to see the actual tests. Can we simplify the result checks, or possibly utilize some helper functions instead? E.g. `expectedResult(div, 1, 2, 0, 5)`

Reply:
For the tests to be as clear and easy as possible, I think it is a good idea to have the whole expected object. If utilities were to generate the expected objects for us, it would be many of them and the parameters would be unclear anyways. I think it is a good idea to sacrifice some lines and readability for clarity. Most editors also support collapsing code blocks which can be handy when dealing with this file.